dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/RequestPriorityAggregator.kt

```diff
@@ -0,0 +1,106 @@
+package org.dataland.datalandcommunitymanager.services
+
+import org.dataland.datalandbackend.openApiClient.model.DataTypeEnum
+import org.dataland.datalandcommunitymanager.model.dataRequest.AggregatedDataRequest
+import org.dataland.datalandcommunitymanager.model.dataRequest.AggregatedDataRequestWithAggregatedPriority
+import org.dataland.datalandcommunitymanager.model.dataRequest.AggregatedRequestPriority
+import org.dataland.datalandcommunitymanager.model.dataRequest.RequestPriority
+
+/**
+ * Implementation of a request priority aggregator to determine the aggregated priority
+ * and filter for a certain aggregated priority
+ */
+class RequestPriorityAggregator {
+    /**
+     * This method determines the aggregated priorities of aggregated data requests.
+     * @param aggregatedDataRequests the aggregated data requests
+     * @returns aggregated data requests with their aggregated priority
+     */
+    fun aggregateRequestPriority(aggregatedDataRequests: List<AggregatedDataRequest>): List<AggregatedDataRequestWithAggregatedPriority> {
+        val aggregatedMap = createAggregatedMap(aggregatedDataRequests)
+        return aggregatedMap
+            .map { (key, priorityCounts) ->
+                createAggregatedDataRequestWithPriority(key, priorityCounts)
+            }
+    }
+
+    /**
+     * Creates a map where the key is a dataset defined by data type, reporting period, and company ID.
+     * The value is a map of request priorities with their respective counts.
+     *
+     * @param aggregatedDataRequests the list of aggregated data requests to process
+     * @returns a map of request counts for each dataset
+     */
+    private fun createAggregatedMap(
```

Comment:
If I understand the code correctly, you count how often a request has a certain priority. In particular, the request counts are not "aggregated" in the sense of aggregated priorities yet. Thus, I propose something like 
```suggestion
    private fun createMapOfCumulatedPriorities(
```
or something a bit less misleading. (Effectively, I wondered reading the code what role the request priority plays here because summing up the numbers happens for each priority individually)

Reply:
I agree